rocketpy/plots/rocket_plots.py

```diff
@@ -225,113 +241,105 @@ def draw(self, vis_args=None):
-                )
-                # close the nosecone
-                ax.plot(
-                    [x_nosecone[-1], x_nosecone[-1]],
-                    [y_nosecone[-1], -y_nosecone[-1]],
-                    color=vis_args["nose"],
-                    linewidth=vis_args["line_width"],
-                )
-
-                # Add the nosecone to the list of drawn surfaces
-                drawn_surfaces.append(
-                    (surface, x_nosecone[-1], surface.rocket_radius, x_nosecone[-1])
-                )
-
+                self._draw_nose_cone(ax, surface, position, drawn_surfaces, vis_args)
             elif isinstance(surface, Tail):
-                x_tail = -csys * surface.shape_vec[0] + position
-                y_tail = surface.shape_vec[1]
-
-                ax.plot(
-                    x_tail,
-                    y_tail,
-                    color=vis_args["tail"],
-                    linewidth=vis_args["line_width"],
-                )
-                ax.plot(
-                    x_tail,
-                    -y_tail,
-                    color=vis_args["tail"],
-                    linewidth=vis_args["line_width"],
-                )
-                # close above and below the tail
-                ax.plot(
-                    [x_tail[-1], x_tail[-1]],
-                    [y_tail[-1], -y_tail[-1]],
-                    color=vis_args["tail"],
-                    linewidth=vis_args["line_width"],
-                )
-                ax.plot(
-                    [x_tail[0], x_tail[0]],
-                    [y_tail[0], -y_tail[0]],
-                    color=vis_args["tail"],
-                    linewidth=vis_args["line_width"],
-                )
+                self._draw_tail(ax, surface, position, drawn_surfaces, vis_args)
+            elif isinstance(surface, Fins):
+                self._draw_fins(ax, surface, position, drawn_surfaces, vis_args)
+        return drawn_surfaces
+
+    def _draw_nose_cone(self, ax, surface, position, drawn_surfaces, vis_args):
+        """Draws the nosecone and saves the position of the points of interest
+        for the tubes."""
+        x_nosecone = -self.rocket._csys * surface.shape_vec[0] + position
+        y_nosecone = surface.shape_vec[1]
+        ax.plot(
+            x_nosecone,
+            y_nosecone,
+            color=vis_args["nose"],
+            linewidth=vis_args["line_width"],
+        )
+        ax.plot(
+            x_nosecone,
+            -y_nosecone,
+            color=vis_args["nose"],
+            linewidth=vis_args["line_width"],
+        )
+        # close the nosecone
+        ax.plot(
+            [x_nosecone[-1], x_nosecone[-1]],
+            [y_nosecone[-1], -y_nosecone[-1]],
+            color=vis_args["nose"],
+            linewidth=vis_args["line_width"],
+        )
+        # Add the nosecone to the list of drawn surfaces
+        drawn_surfaces.append(
+            (surface, x_nosecone[-1], surface.rocket_radius, x_nosecone[-1])
+        )
```

Comment:
This is strange to me. You add the surface to a list, which was passed as an argument to the function. If you don't return the list with the added surfaces, is this code line really necessary? I'm afraid that everytime you are starting with an empty list.

```
        # Add the nosecone to the list of drawn surfaces
        drawn_surfaces.append(
            (surface, x_nosecone[-1], surface.rocket_radius, x_nosecone[-1])
        )
```

Reply:
This behavior is also present in other parts of the code.

Reply:
The `drawn_surfaces` list is created and then appended to the various draw methods I created. That same list is used later to draw the tubes later.

Remember that a list in Python is just a pointer. Changing the name of the list variable does not change the address that it points to.